apps/api/v2/src/modules/organizations/controllers/event-types/organizations-event-types.e2e-spec.ts

```diff
@@ -0,0 +1,458 @@
+        });
+    });
+
+    it("should assign all members to managed event-type", async () => {
+      const body: UpdateTeamEventTypeInput_2024_06_14 = {
+        assignAllTeamMembers: true,
+      };
+
+      return request(app.getHttpServer())
+        .patch(`/v2/organizations/${org.id}/teams/${team.id}/event-types/${managedEventType.id}`)
+        .send(body)
+        .expect(200)
+        .then(async (response) => {
+          const responseBody: ApiSuccessResponse<TeamEventTypeOutput_2024_06_14[]> = response.body;
+          expect(responseBody.status).toEqual(SUCCESS_STATUS);
+
+          const data = responseBody.data;
+          expect(data.length).toEqual(3);
+
+          const teammate1EventTypes = await eventTypesRepositoryFixture.getAllUserEventTypes(teammate1.id);
+          const teammate2EventTypes = await eventTypesRepositoryFixture.getAllUserEventTypes(teammate2.id);
+          const teamEventTypes = await eventTypesRepositoryFixture.getAllTeamEventTypes(team.id);
+          const managedTeamEventTypes = teamEventTypes.filter(
+            (eventType) => eventType.schedulingType === "MANAGED"
+          );
+
+          expect(teammate1EventTypes.length).toEqual(1);
+          expect(teammate2EventTypes.length).toEqual(1);
+          expect(managedTeamEventTypes.length).toEqual(1);
+          expect(managedTeamEventTypes[0].assignAllTeamMembers).toEqual(true);
+
+          const responseTeamEvent = responseBody.data[0];
+          expect(responseTeamEvent?.teamId).toEqual(team.id);
+          expect(responseTeamEvent.assignAllTeamMembers).toEqual(true);
+
+          const responseTeammate1Event = responseBody.data[1];
+          expect(responseTeammate1Event?.ownerId).toEqual(teammate1.id);
+          expect(responseTeammate1Event?.parentEventTypeId).toEqual(responseTeamEvent?.id);
+
+          const responseTeammate2Event = responseBody.data[2];
+          expect(responseTeammate2Event?.ownerId).toEqual(teammate2.id);
+          expect(responseTeammate2Event?.parentEventTypeId).toEqual(responseTeamEvent?.id);
+
+          managedEventType = responseTeamEvent;
+          managedTeammate1EventType = responseTeammate1Event;
+          managedTeammate2EventType = responseTeammate2Event;
+        });
+    });
+
+    it("should delete collective event-type", async () => {
+      return request(app.getHttpServer())
+        .delete(`/v2/organizations/${org.id}/teams/${team.id}/event-types/${collectiveEventType.id}`)
+        .expect(200);
+    });
+
+    it("should delete managed event-type", async () => {
+      return request(app.getHttpServer())
+        .delete(`/v2/organizations/${org.id}/teams/${team.id}/event-types/${managedEventType.id}`)
+        .expect(200);
+    });
+
+    function evaluateHost(expected: Host, received: Host) {
+      expect(expected.userId).toEqual(received.userId);
+      expect(expected.mandatory).toEqual(received.mandatory);
+      expect(expected.priority).toEqual(received.priority);
+    }
```

Comment:
can we add tests for some fail case ? like trying to update an existing event-type that does not belong to the team ?

Reply:
added in [fix: dont allow creating event-type for non-team member](https://github.com/calcom/cal.com/pull/15775/commits/b371e72fb9206499bf59873c3abe199ed859e48a) and [tests: negative tests](https://github.com/calcom/cal.com/pull/15775/commits/913d43bb5ed551f8b5aae675f07c1b349c68aac0). All tests can be seen in [organizations-event-types.e2e-spec.ts](https://github.com/calcom/cal.com/pull/15775/files#diff-293a7771b8112204ce3086bb48fa224876fce328e8a8dc8eb7227b60d57efb35)